ocean_lib/web3_internal/contract_base.py

```diff
@@ -210,7 +215,19 @@ def send_transaction(
             _transact.update(transact)
 
         # TODO: transfer rest of custom contract function, waiting etc.
-        return getattr(self.contract, fn_name)(*fn_args, _transact).txid
+        tx_id = getattr(self.contract, fn_name)(*fn_args, _transact).txid
+        tx_receipt = wait_for_transaction_receipt_and_block_confirmations(
```

Comment:
I wonder if we should hold off on any work in this PR, until the brownie stuff goes in. Brownie's gonna mean big changes in how we handle base_contract, etc.

Reply:
since this branch is based off of ocean-brownie and it's just an exploratory branch to check tests, I think we can use this setup. True, we have other priorities, but it won't be double the work if we tweak it now.